beatrix/src/test/java/org/killbill/billing/beatrix/integration/TestCatalogFixedTermWithVersionedCatalogs.java

```diff
@@ -0,0 +1,132 @@
+    public void beforeMethod() throws Exception {
+        if (hasFailed()) {
+            return;
+        }
+
+        super.beforeMethod();
+
+        // Setup tenant
+        clock.setTime(new DateTime("2021-01-01T12:56:02"));
+        testCallContext = setupTenant();
+
+        // Setup account in right tenant
+        account = setupAccount(testCallContext);
+        
+    }
+
+
+    @Test(groups = "slow")
+    public void testVersionedCatalogsEffectiveDateBetweenFixedTerm() throws Exception {
+
+        busHandler.pushExpectedEvents(NextEvent.CREATE, NextEvent.BLOCK, NextEvent.INVOICE, NextEvent.INVOICE_PAYMENT,
+                NextEvent.PAYMENT);        
+        final PlanPhaseSpecifier spec = new PlanPhaseSpecifier("plumber-insurance-monthly", null);
+        final UUID entitlementId = entitlementApi.createBaseEntitlement(account.getId(), new DefaultEntitlementSpecifier(spec, null, UUID.randomUUID().toString(), null), "something", null, null, false, true, ImmutableList.<PluginProperty>of(), testCallContext);
+        assertNotNull(entitlementId);
+        assertListenerStatus();
+        invoiceChecker.checkInvoice(account.getId(), 1, testCallContext, new ExpectedInvoiceItemCheck(new LocalDate(2021, 01, 01), new LocalDate(2021, 02, 01), InvoiceItemType.RECURRING, new BigDecimal("40.00")));
+        
+        busHandler.pushExpectedEvents(NextEvent.INVOICE, NextEvent.PAYMENT, NextEvent.INVOICE_PAYMENT);
+        clock.addMonths(1); //2021-02-01
+        assertListenerStatus();
+        invoiceChecker.checkInvoice(account.getId(), 2, testCallContext, new ExpectedInvoiceItemCheck(new LocalDate(2021, 02, 01), new LocalDate(2021, 02, 15), InvoiceItemType.RECURRING, new BigDecimal("20.00")));
+        
+        //price change takes effect
+        busHandler.pushExpectedEvents(NextEvent.INVOICE, NextEvent.PAYMENT, NextEvent.INVOICE_PAYMENT);
+        clock.addDays(15); //2021-02-15
+        assertListenerStatus();
+        invoiceChecker.checkInvoice(account.getId(), 3, testCallContext, new ExpectedInvoiceItemCheck(new LocalDate(2021, 02, 15), new LocalDate(2021, 03, 01), InvoiceItemType.RECURRING, new BigDecimal("30.00")));
+        
+        busHandler.pushExpectedEvents(NextEvent.INVOICE, NextEvent.PAYMENT, NextEvent.INVOICE_PAYMENT);
+        clock.addDays(15); //2021-03-01
+        assertListenerStatus();
+        invoiceChecker.checkInvoice(account.getId(), 4, testCallContext, new ExpectedInvoiceItemCheck(new LocalDate(2021, 03, 01), new LocalDate(2021, 04, 01), InvoiceItemType.RECURRING, new BigDecimal("60.00")));
+        
+        busHandler.pushExpectedEvents(NextEvent.INVOICE, NextEvent.PAYMENT, NextEvent.INVOICE_PAYMENT);
+        clock.addMonths(1); //2021-04-01
+        assertListenerStatus();
+        invoiceChecker.checkInvoice(account.getId(), 5, testCallContext, new ExpectedInvoiceItemCheck(new LocalDate(2021, 04, 01), new LocalDate(2021, 04, 11), InvoiceItemType.RECURRING, new BigDecimal("20.00")));
```

Comment:
Perhaps add a comment that 100 days (length of the fixed term) from Jan 1st leads to April 11 (year 2021)

Reply:
Fixed